atomicBool/atomicBool.go

```diff
@@ -0,0 +1,35 @@
+package atomicBool
+
+import (
+	"sync/atomic"
+	"unsafe"
+)
+
+// Bool is an atomic bool type
+type Bool int32
```

Comment:
I was playing around with it a bit, I originally had it as Set(state bool). For some reason b.True()/False felt cleaner to use. 

I'm not completely opposed to Set(state) though. @shahzilabid - can we get your input?